### ISSUE TITLE
Display all notes from a folder (eg. Daily notes)?

### ISSUE THREAD

**Issue by sbsamarski** [gasparschott/obsidian-continuous-mode#4]
This looks great!

A suggestion - is it possible to add a functionality to display all notes from a certain folder with some date filter? For example display all notes from Daily notes folder from February 2024?

**Comment by gasparschott** [comment]
Thanks. I considered adding this sort of functionality, but I thought it was beyond the scope of the plugin. I am torn, however, because I certainly recognize its utility; I might end up adding an open-all-files-in-folder function at some point.

Meanwhile, the "Weekly Review" plugin might be close to what you want: https://github.com/brandonkboswell/weekly-review

**Comment by sbsdevplus** [comment]
Thanks for the suggestion. I looked up the Weekly review plugin, but it just opens a bunch of daily notes, which isn't what I need.
Your plugin is EXACTLY what I (and many others) need. If you search you'll see this question and the need being expressed by various users for a few years already, and no one can find anything good.
I'm almost manually writing and adding ![[2024-02-xx]] lines in an "Daily notes overview" note, one below the other, just to have this functionality, lol!

The thing is that almost everyone writes daily notes, so a plugin like this that gives an overview of let's say last x daily notes could easily end up being EXTREMELY popular. It's definitely in demand, I can say that.

**Comment by gasparschott** [comment]
Funnily enough, I don't use daily notes at all. Can you describe how you imagine this functionality would work? Right-click on a folder in the file explorer—and then what? Some sort of file-picker/filter window appears? Would the filter be looking at front-matter, file name, file metadata, or...? 

It still seems to me that a separate plugin would be a better fit for this, though. I think maybe one reason such a plugin doesn't exist (yet?) is that it could be possible to mistakenly open hundreds or thousands of notes at a time, which would be a bit of a mess.

**Comment by daxmonson** [comment]
I've thought about and wanted this too and I think you're right in it hasn't been developed because of that last paragraph. 

Maybe it would be possible to have a modal pop-up that you could specify like 1-15, 15-30 and so on of the notes in folder. Can understand that building it out would be tricky and might be out of the scope of what you're interested in. Would be cool to see though.

**Comment by sbsdevplus** [comment]
I think it could be made that whenever you click on some button (ribbon or action or whatever) a note or a window opens with the list of last x notes. You could just make a setting in plugin settings where a user can choose how many notes to list. I think that would be simple and to the point, avoiding any problems.
Or a question appears every time asking for how many notes to display, where you could limit it to let's say 100 (daily notes are usually pretty simple).
In the settings you could make an option to choose whether to include front matter in the list or not. But the note name should, of course, be displayed on top.

As for how and where to look for info - daily notes are sorted by file names (usually "yyyy-mm-dd"), so it's basically just listing the last x notes in a folder, sorted by name DESC. As simple as that. Or you could add a setting to choose the sorting option.

If you want to make more options later on you could add an option whether or not to include subfolders, for those who use weekly or monthly or yearly subfolders. So the plugin would list let's say last 50 notes (sorted by name DESC) from one folder, or include all subfolders.

Later on, if you want to expand on the plugin you could add all kinds of filtering options, to look into the front matter or metadata or tags, and include only selected filter options.

Or later-later you could define something like a dataview block that lists entire notes one after another, based on some basic filters (doesn't have to be too smart - it could have just basic options for sort by asc/desc, (not) include folder, (not) include tag, and/or with parenthesis).
Basically like a super basic dataview list, but that displays entire notes, not just file names.

Something like an "Note preview list"...

**Comment by salehm3** [comment]
This same functionality for folders is very much needed!!!

**Comment by gasparschott** [comment]
I'm considering adding this or something similar—perhaps an "Open selected notes in Continuous Mode" contextual menu item—but my thought is still that this functionality should be a separate plugin. Stay tuned.

**Comment by sbsamarski** [comment]
I would suggest to make it automatic. For example to set certain parameters in settings, and the user can just click on a button (easy to make with Commander plugin), which will list a certain number of notes in continuous mode. So, to be a one click command that opens certain number of notes.
If one has to select notes each time then it will become tiresome after some time...
That's why the dataview plugin is so popular - it can make many everything automatic.

And for the settings there could be for example max number of notes to list, a folder, sorting order (time asc, time desc, name asc, name desc). That's all.

And then later if you want to improve the plugin you could add an option to add multiple settings for each folder, so a user can configure multiple buttons if needed.

**Comment by daxmonson** [comment]
"Open selected notes in Continuous Mode contextual menu item" that's probably the more elegant solution. If you do this in continuous mode or a separate plugin, I hope there's a way to multi-select that doesn't involve a keyboard having to be attached on mobile/tablet for shift-select. Actually, building something for Obsidian too and lack of keyboard modifiers on the ios keyboard is a pain point.

**Comment by a3Sun** [comment]
I think selected is also a good use case but personally would realllyyyyy love functionality anllowing automatic continuous mode of all other notes in an opened notes folder.

**Comment by gasparschott** [comment]
I'm curious as to how you all think that Continuous Mode should handle opening a second (or third, etc.) folder of notes?\* Should it automatically open the notes in a new split? Should it replace all the notes in the currently active split with the new batch? Should there be an option to append notes in the currently active split instead?

\* Assuming I go ahead with adding this functionality :-)

@sbsamarski: Those are interesting suggestions. Certainly a max number of notes to open would be useful, or at least a warning. Note that a present the plugin is really rather simple: most of what it does is accomplished via basic CSS. Things like sorting notes and accessing metadata, etc. are likely to come later than sooner.

@daxmonson: I'll look into selecting multiple notes in the mobile app File Explorer without a keyboard. Somehow it seems unlikely to be possible, but I haven't used Obsidian much on my iPad.

@a3Sun: Are you suggesting something like clicking a folder in the File Explorer should automatically open all its notes in Continuous Mode? Or do you mean that when any note in a folder is opened, all its sibling notes should automatically open as well? In either case, the answers to my questions above are important.

**Comment by a3Sun** [comment]
I was imagining it as when you select a note, all sibling notes automatically being open in the order of the folder structure. However selecting the folder is certainly not a bad idea, I just personally prefer the former to help me stay exposed to notes I might forget about and make new associations/ connections. Just would make obsidian very exploratory. I don’t think nested folders should be displayed by default, allow the option to append though. I wonder if there’s any way to visually show the folders so you can keep on ‘exploring’ after you reach the end of a continuous folder. I’m not sure how hard any of this is to implement, personally I’m happy with almost any variation, but this would be my preference.

**Comment by sbsamarski** [comment]
I would insert an option in the right click menu. 
So if you right click on a folder, select the option to open the (already defined) number of notes from that folder. The sorting order of the notes could match the displayed folder. Or maybe even better - when you select the command have a small popup question/dialog box asking for the number of notes to display and the sorting order (with options for time and name, both asc and desc). And maybe a check box with "open in new tab" (when unchecked it'll open the list in the current tab). 

Of course, limit the number of notes to some normal number, so Obsidian doesn't crash if there are too many notes (if there's a possibility of something like that). Or make a warning if a user selects let's say more than 100 notes. (Other notes might be longer, but daily notes are usually quite short, so shouldn't be a problem.)

Additionally, you could also have an option to right click on an individual note, with the same popup question, which would open the list starting from THAT note, whatever the sorting order is selected.

So, right click on a folder and right click on a note, would give very customizable lists, one that starts from the beginning of the folder, and one that starts from the selected note.

You could either put settings for the number of notes, the sorting order and "open in new tab" option in Obsidian Plugin settings, or as a popup that comes up when you launch the command. Whatever is easier for you, I'd be OK with either.

**Comment by sbsamarski** [comment]
Just to add - if you want to keep it super simple and without ANY settings or popup questions, then you could just display all the notes from the selected folder (with a hard limit of 100 notes, just in case there are too many), sorted in the same way as the folder is displayed. 

That's all. And yes, I personally think it would be better to open the list in its own, new tab.

**Comment by gasparschott** [comment]
Check out the latest version (1.2.0) just posted.

**Comment by sbsamarski** [comment]
The plugin got automatically uninstalled with the latest Obsidian update, and if I try to install it, it says "failed to install plugin".

It looks like something changed with the new version of Obsidian, which made the plugin incompatible.

**Comment by gasparschott** [comment]
I fixed the issue—there was a mismatch between the version number and the Github tag number.

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by danieltomasz** [comment]
Thanks for the update @gasparschott ! One thing I noticed that when the file is move to other folder, it stays in tabbed view, but this is not a bug but rather consequence that  the view is per tab view, and the plugin doesn't have any other way to track which files it should display

**Comment by gasparschott** [comment]
Thanks, this situation is not something I had considered before. I might see if there is a way to handle it, but it seems likely it would require a significant rethink of the plugin functions, which is not something I want to deal with just yet. If it were implemented, then I think that there would be an equal expectation that notes dragged into the continuous mode folder would also appear in the tab group.
